components/public-api/README.md

```diff
@@ -0,0 +1,101 @@
+# Public-API
+
+# Example Flows
+This section gives examples how clients would use this API.
+
+## Create Workspace
+
+### Not prebuild aware
+```go
+workspaces.CreateWorkspace({
+    idempotency_token: "<random_string>",
+    context_url: "https://github.com/gitpod-io/gitpod",
+})
+```
+
+### Prebuild aware but no log support
+```go
+workspaces.CreateWorkspace({
+    idempotency_token: "<random_string>",
+    context_url: "https://github.com/gitpod-io/gitpod",
+    prebuild: {
+        if_available: true,
+    }
+})
+```
+
+### Prebuild aware with log support
+```go
+contextURL := "https://github.com/gitpod-io/gitpod"
+prb := prebuilds.GetRunningPrebuild({ context_url: contextURL })
```

Comment:
I believe starting prebuilds and starting workspaces have so much in common, that we should rather generalize over them and make 'prebuild' a property on the workspaces API rather than duplicating everything in a prebuilds API.

Reply:
At least for listening to logs and status updates, we could use a prebuild's workspace instance. Also there might be an image build ongoing.